### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by anjan-keysight**
Data added for minimum resource requirements.

### CODE REVIEW

**Comment by anjan-keysight** [comment]
@bortok @apratimmukherjee I have updated README with minimum resource requirements, kindly review.

File: README.md [hunk 1]
```diff
@@ -95,6 +95,94 @@ status:
   state: DEPLOYED
 ```
 
+## Minimum resource requirement
+
+The minimum memory and cpu requirements for each ixia-c components are captured in the following table. Kubernetes metrics server has been used to collect resource usage data. The memory represents the minimum working set memory required, and for protocol and traffic engines, varies depending on the number of colocated ports. The figures are in Mi or MB.
```
Comment (apratimmukherjee) [inline README.md:100]: Note sure why some words are showing up in red.

Comment (apratimmukherjee) [inline README.md:100]: number of co-located ports e.g. when multiple ports are added to a 'group' for LAG use-cases when a single test container has more than 1 test NIC connected to the DUT.

Comment (apratimmukherjee) [inline README.md:100]: The figures are in Mi or MB per container and does not include shared or cached memory across multiple containers/pods in a system.

File: README.md [hunk 2]
```diff
@@ -95,6 +95,94 @@ status:
...
+The cpu resource figures are in millicores.
+
+                | ---------- | -------- | ------- | ---------- | ----- |
+                |            | Protocol | Traffic | Controller | gNMI  |
+                |            | Engine   | Engine  |            |       |
+                | ---------- | -------- | ------- | ---------- | ----- |
+                | Min CPU    |   200    |   200   |     10     |   10  |
+                | ---------- | -------- | ------- | ---------- | ----- |
+
+The operator sets the minimum memory requirement for each ixia-c component depending on the port configuration. The minimum cpu requirement is set to a default value for each component.
```
Comment (apratimmukherjee) [inline README.md:124]: to the default value for each component as shown in the table above.

Comment (apratimmukherjee) [reply README.md:124]: depending on the port configuration similar to what is shown in the table showing minimum memory requirements in different deployment configurations.

File: README.md [hunk 3]
```diff
@@ -95,6 +95,94 @@ status:
...
+                | Component       | 1 Port  | 2 Port  | 4 Port  | 6 Port  | 8 Port  |
+                |                 | Default |         |         |         |         |
+                | --------------- | ------- | ------- | ------- | ------- | ------- |
+                | Protocol Engine |   350   |   420   |   440   |   460   |   480   |
+                | --------------- | ------- | ------- | ------- | ------- | ------- |
+                | Traffic Engine  |    60   |    70   |    90   |   110   |   130   |
+                | --------------- | ------- | ------- | ------- | ------- | ------- |
+                | Controller      |    25   |         |         |         |         |
+                | --------------- | ------- | ------- | ------- | ------- | ------- |
+                | gNMI            |    15   |         |         |         |         |
```
Comment (apratimmukherjee) [inline README.md:112]: Maybe a * can be added with a clarification that gNMI and Controller have a fixed minimum memory requirement and is currently not dependent on number of test ports for the topology.

File: README.md [hunk 4]
```diff
@@ -95,6 +95,94 @@ status:
...
+                | ---------- | -------- | ------- | ---------- | ----- |
+                |            | Protocol | Traffic | Controller | gNMI  |
+                |            | Engine   | Engine  |            |       |
+                | ---------- | -------- | ------- | ---------- | ----- |
+                | Min CPU    |   200    |   200   |     10     |   10  |
+                | ---------- | -------- | ------- | ---------- | ----- |
+
+The operator sets the minimum memory requirement for each ixia-c component depending on the port configuration. The minimum cpu requirement is set to a default value for each component.
+
+### Minimum resource usage based on various test configurations
```
Comment (apratimmukherjee) [inline README.md:126]: Minimum and maximum

File: README.md [hunk 5]
```diff
@@ -95,6 +95,94 @@ status:
...
+
+Depending on the nature of test run, the memory and cpu resource requirements may vary across all ixia-c components. The following tables capture the memory usage for lag scenarios with varying numbers of member ports. The minimum value represents initial memory on topology deployment and maximum value indicates the peak memory usage during the test run. Figures are in Mi or MB.
```
Comment (apratimmukherjee) [inline README.md:128]: lag -> LAG

Comment (apratimmukherjee) [inline README.md:128]: table captures

**Review by apratimmukherjee — APPROVED** [review]
Overall looks good to me. Few comments provided.

File: README.md (hunk 1 above)
Comment (anjan-keysight) [reply README.md:100]: Looks fine if you go through 
https://github.com/open-traffic-generator/ixia-c-operator/tree/dev-resource-reqm

Comment (anjan-keysight) [reply README.md:100]: Updated

Comment (anjan-keysight) [reply README.md:100]: Updated.

File: README.md (hunk 2 above)
Comment (anjan-keysight) [reply README.md:124]: Modified.

File: README.md (hunk 3 above)
Comment (anjan-keysight) [reply README.md:112]: Added footnote

File: README.md (hunk 4 above)
Comment (anjan-keysight) [reply README.md:126]: Updated

File: README.md (hunk 5 above)
Comment (anjan-keysight) [reply README.md:128]: Updated.

Comment (anjan-keysight) [reply README.md:128]: Updated

**Review by bortok — CHANGES_REQUESTED** [review]
Hi Anjan,

I believe the min requirements values don't belong to the operator repository, and instead should be documented in Ixia-c repo, probably here https://github.com/open-traffic-generator/ixia-c/blob/mkdocs/docs/prerequisites.md. You could reference that documentation from this readme and explain the operator is honoring these numbers.

**Comment by anjan-keysight** [comment]
@bortok I have updated the ixia-c prerequisites directly, and added link from this README, as you suggested.

**Comment by bortok** [comment]
@anjan-keysight high chance we will forget to update the link to point to `main` once `mkdocs` branch is merged. Could you please track it so that we don't? Otherwise, I'm good.

**Review by bortok — APPROVED** [review]

**Comment by anjan-keysight** [comment]
I will keep note and update again when mkdocs is merged.

**Pull request merged**